tmt/steps/prepare/ansible.py

```diff
@@ -165,6 +165,9 @@ def go(
 
                 logger.info('playbook-path', playbook_path, 'green')
 
-            guest.ansible(playbook_path, extra_args=self.data.extra_args)
+            guest.ansible(
+                playbook_path,
+                playbook_root=self.step.plan.fmf_root,
```

Comment:
I believe this would bring it closer to addressing #2970, #2634. To that extent, I believe that we want to make the `playbook_root` point to a persistent path for all `provision` steps [^1], that way it can be reused by other guests. At the very least the inventory part should point there, but I believe roles, ansible-galaxy, and such are also valuable to be placed in a reusable path.

But I kinda see why you want to point it to `fmf_root` since you want to move the configurations from the fmf tree to `playbook_root`. Can you cook up a simple fmf tree to show how you want to reuse references from `fmf_root` in this scenario? Maybe we can think of an equivalent interface for this?

[^1]: https://github.com/teemtee/tmt/issues/2634#issuecomment-1905996432

Reply:
It's my quick draft from yesterday which @falconizmi plans to finish, a few sentences about the motivation: fmf root gets in because of the "user-provided playbooks shall come from fmf root and nowhere else", to avoid `playbook: /usr/share/very-secret-erase-business-data.yaml` cases.

I'm not sure we could, to prevent the user from sneaking around the runner FS, replace "playbook root" with anything but the fmf root. The pre-generated, shared, and persistent inventory and host vars should be orthogonal to this problem, we still can - and should - have both at the end, plus collections and even more. Bear in mind that this PR is supposed to just do a small clean-up to unblock (correct) use of playbooks bundled with tmt & other @falconizmi's work, it probably shouldn't turn into something much bigger, but if it sparks more patches improving the Ansible situation, nobody will complain, especially not me :)

Reply:
Ah, ok, didn't know `ansible-playbook` had a default search path for resolving playbooks, didn't quite find documentation on that. But if that's the main concern, wouldn't it be equivalent to resolve `playbook_path` to an absolute path, maybe with a check for `.is_relative_to(`/usr`)`?

But what about safeguarding against `ansible.builtin.import_playbook`? Wouldn't the user also be able to accidentally/intentionally run a protected playbook from there? Maybe setting up an `ansible.cfg` is also needed?